### DISCUSSION TITLE
Drawing rectangle on the chart

### DISCUSSION THREAD

**Discussion by quankhuc** [tradex-app/TradeX-chart#38] [Q&A]
Hi,

I have tried all of your examples anad it works really well. Thanks for making an awesome library and share it with the community. I learned a lot from this repo. However, I want to have a toolbar where I can draw a rectangle on the chart. Is there a way to do that yet or is it a feature being developed? I am using nextJS

**Comment by tradex-app** [comment]
Hi Quan, thank you! =)

Yes, drawing tools are on the development road map. The tool bar exists, but currently has no functioning tools.

Pixel perfect hit detection is currently under development. This will allow the detection of mouse clicks on chart drawings. Once that is complete, then work can proceed with implementing the drawing tools.

There is a Discord TradeX group where news and updates are posted.

https://discord.gg/ByQV3UWAK

**Comment by tradex-app** [comment]
Here's the development road map:

https://tradex-app.github.io/TradeX-chart/TradeX-chart-Development-Roadmap.pdf

**Comment by gmalnk** [comment]
Hi,
Any updates on the drawing tool development? I'd like to use it for my project, especially with the ability to draw trendlines on the charts.

**Comment by tradex-app** [comment]
Hi @gmalnk, the ground work is done. I need to find / make time to work on the drawing tools.
I've had to focus on other projects that pay my bills and hence the chart development has slowed down these past months.
News about the chart is updated more regularly on Discord.
https://discord.gg/6XS9tDrcdq
